### PR TITLE
[FLINK-23916][python][docs] Expose in python and update the documentation how to set tolerable checkpoint failure number

### PR DESCRIPTION
**Pull request by pnowojski**
This PR adds python API and updates the docs for setting the tolerable number of checkpoint failures.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**yes** / no)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / **JavaDocs** / not documented)

### CODE REVIEW

File: flink-python/pyflink/datastream/checkpoint_config.py [hunk 1]
```diff
@@ -214,6 +214,35 @@ def set_fail_on_checkpointing_errors(self,
...
+                                                ) -> 'CheckpointConfig':
+        """
+        This defines how many consecutive checkpoint failures will be tolerated, before the whole
+        job is failed over. The default value is `0`, which means no checkpoint failures will be
+        tolerated, and the job will fail on first reported checkpoint failure.
+
+        Example:
+        ::
+
+            >>> config.set_tolerable_checkpoint_failure_number(False)
```
Comment (WeiZhong94) [inline flink-python/pyflink/datastream/checkpoint_config.py:237]: The param should be an `int`?

Comment (pnowojski) [reply flink-python/pyflink/datastream/checkpoint_config.py:237]: Good catch! A copy paste error. Fixed.

**Comment by WeiZhong94** [comment]
Thanks for your update. The Python part looks good now.  :)

**Comment by curcur** [comment]
LGTM,

**Review by curcur — APPROVED** [review]

**Comment by pnowojski** [comment]
Thanks @curcur, I've created a ticket for that:
https://issues.apache.org/jira/browse/FLINK-24155

**Pull request merged**